internal/services/ingest.go

```diff
@@ -16,25 +17,75 @@ import (
 	"github.com/stellar/wallet-backend/internal/utils"
 )
 
-type IngestManager struct {
-	PaymentModel      *data.PaymentModel
-	LedgerBackend     ledgerbackend.LedgerBackend
-	NetworkPassphrase string
-	LedgerCursorName  string
-	AppTracker        apptracker.AppTracker
+type IngestService interface {
+	Run(ctx context.Context, start, end uint32) error
 }
 
-func (m *IngestManager) Run(ctx context.Context, start, end uint32) error {
+var _ IngestService = (*ingestService)(nil)
+
+type ingestService struct {
+	models            *data.Models
+	ledgerBackend     ledgerbackend.LedgerBackend
+	networkPassphrase string
+	ledgerCursorName  string
+	appTracker        apptracker.AppTracker
+	httpClient        utils.HTTPClient
+	rpcURL            string
+}
+
+func NewIngestService(
+	models *data.Models,
+	ledgerBackend ledgerbackend.LedgerBackend,
+	networkPassphrase string,
+	ledgerCursorName string,
+	appTracker apptracker.AppTracker,
+	httpClient utils.HTTPClient,
+	rpcURL string,
+) (*ingestService, error) {
+	if models == nil {
+		return nil, errors.New("models cannot be nil")
+	}
+	if ledgerBackend == nil {
+		return nil, errors.New("ledgerBackend cannot be nil")
+	}
+	if networkPassphrase == "" {
+		return nil, errors.New("networkPassphrase cannot be nil")
+	}
+	if ledgerCursorName == "" {
+		return nil, errors.New("ledgerCursorName cannot be nil")
+	}
+	if appTracker == nil {
+		return nil, errors.New("appTracker cannot be nil")
+	}
+	if httpClient == nil {
+		return nil, errors.New("httpClient cannot be nil")
+	}
+	if rpcURL == "" {
+		return nil, errors.New("rpcURL cannot be nil")
+	}
+
+	return &ingestService{
```

Comment:
how are you planning on making rpc calls inside the ingestor? Won't you need the transaction service for it?

Reply:
So then will the implementation of getLedgerEntries come in a following pr?